pythonfuzz/tracer.py

```diff
@@ -0,0 +1,23 @@
+import collections
+import sys
+
+prev_line = 0
+data = collections.defaultdict(set)
+
+def trace(frame, event, arg):
+    if event != 'line':
+        return trace
+    global prev_line
+    func_filename = frame.f_code.co_filename
+    func_line_no = frame.f_lineno
+
+    data[func_filename].add((prev_line, func_line_no))
```

Comment:
This might miss cases in inter-file control transfer.

If you're in file `bob.py` at line 23 and you call a function, it'll look exactly the same as being in `alice.py` as line 23 calling the same function. It's not a likely case, but it's one that could easily miss the case where you make progress in one file, mistaking it for another.